packages/ra-no-code/src/ui/ImportResourceDialog.tsx

```diff
@@ -52,7 +53,7 @@ export const ImportResourceDialog = (props: ImportResourceDialogProps) => {
                     if (resourceAlreadyExists) {
                         // If we imported more records for an existing resource,
                         // we must refresh the list
-                        refresh();
+                        queryClient.refetchQueries([resource, 'getList']);
```

Comment:
Why?

Reply:
Because useGetList isn't affected by refresh anymore (I've removed the version trick). We need to use react-query to refresh the list of records after importing new ones.

Reply:
Ok but why did you remove the version trick? Shouldn't it still be affected by the version?

Reply:
react-query provides a way to force refetch already. We'll remove the version reducer, but first we must make views work without version.

Reply:
Ok but then I think we should find a way to hide this behind a custom hook

Reply:
Yes, we'll rewrite the useRefresh hook but after this PR, which is large enough already.